lib/poparser.js

```diff
@@ -473,15 +543,12 @@ Parser.prototype._finalize = function (tokens) {
  *
  * @constructor
  * @param {String} [defaultCharset] Default charset to use
- * @param {String} [options] Stream options
+ * @param {import('readable-stream').TransformOptions} [options] Stream options
+ * @param {Boolean} [validation=false] Enables validation of parsed content, if set to true
  */
-function PoParserTransform (defaultCharset, options) {
-  if (!options && defaultCharset && typeof defaultCharset === 'object') {
-    options = defaultCharset;
-    defaultCharset = undefined;
-  }
-
+function PoParserTransform (defaultCharset, options, validation) {
```

Comment:
See above. A method signature like `PoParserTransform (options, streamOptions)` would be better I think. Meaning: the current `options` parameter will become `streamOptions` containing only the options passed specifically for `Transform` below.

Reply:
Thank you for your review.

Personally, I am not sure whether having a single `options` argument instead of two dedicated ones would make much difference for a library user. The argument order is still preserved and, hence, is backward compatible. At the same time, I think that from the perspective of project maintainer using control flow directly instead of `unshiftOptionalArgs` would complicate argument parsing quite significant. Here is a list of singatures `stream` function must comply a logic I came up with:

```js
/**
 * Parses a PO stream, emits translation table in object mode
 *
 * @typedef {{ defaultCharset: strubg, validation: boolean }} Options
 * @param defaultCharset Default charset to use
 * @param transformOptions Stream options
 * @param options Optional options with defaultCharset and validation
 * @type {{
*    () => Stream;
*    (defaultCharset: string) => Stream;
*    (defaultCharset: string, transformOptions: import('readable-stream').TransformOptions) => Stream;
*    (transformOptions: import('readable-stream').TransformOptions) => Stream;
*    (options: Options) => Stream;
*    (options: Options, transformOptions: import('readable-stream').TransformOptions) => Stream;
* }}
*/
module.exports.stream = function (arg0, arg1) {
  const [defaultCharset, validation, transformOptions] = typeof arg0 === 'string'
    ? [arg0, undefined, arg1]
    : 'defaultCharset' in arg0
      ? [arg0.defaultCharset, arg0.validation, arg1]
      : [undefined, undefined, arg0];

  return new PoParserTransform(defaultCharset, transformOptions, validation);
};
```

I am not very fond of this approach and I would be curious to learn if there is a better way to solve this. Regardless, if you think the above snippet is the right way to go, please let me know and I'll apply it at once.

Thanks.

Reply:
I might not be skilled enough to have this discussion - my appologies if I'm missing something.
My point of view is: considering plain JS use (not thinking about TS features or JSDoc-based code completion), arguments that can be used in any order (based on their types being different) are confusing. These methods (`parse` and `stream`) will likely get more arguments in the future.
To solve this, an options object is common (as properties are not mandatory and their existence can easily be checked), no?
Or is this an outdated view? Are you aware of other libraries using an `unshiftOptionalArgs` approach (or similar)?

Reply:
I see, thanks for sharing your point of view. I made necessary adjustments so please take another look.

Reply:
Thanks, I think this makes things more simple. I'll write a few (last) comments.